tritium-core/src/main/java/com/palantir/tritium/event/InstrumentationProperties.java

```diff
@@ -80,28 +79,23 @@ public static void reload() {
     }
 
     private static Map<String, String> createInstrumentationSystemProperties() {
-        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
-        copySystemProperties().forEach((key, value) -> {
-            if (String.valueOf(key).startsWith(INSTRUMENT_PREFIX)) {
-                builder.put(String.valueOf(key), String.valueOf(value));
-            }
-        });
-        Map<String, String> map = builder.build();
-        log.debug("Reloaded instrumentation properties {}", map);
-        return map;
-    }
-
-    private static Map<Object, Object> copySystemProperties() {
         /*
-         * Since system properties are a hash table, they can be a point of contention
-         * as all access is synchronized. We therefore take an approach of copying all the
-         * keys at once, but must hold the lock on the entire Hashtable.
+         * Since system properties are backed by a java.util.Hashtable, they can be
+         * a point of contention as all access is synchronized. We therefore take
+         * an approach of cloning the entire Hashtable (which does its own
+         * locking), then copying only the entries we are interested in keeping.
+         *
+         * See https://bugs.openjdk.java.net/browse/JDK-6977738 and https://bugs.openjdk.java.net/browse/JDK-8029891
          */
-        Properties properties = System.getProperties();
-        //noinspection SynchronizationOnLocalVariableOrMethodParameter
-        synchronized (properties) {
-            return ImmutableMap.copyOf(properties);
-        }
+        @SuppressWarnings("unchecked")
+        Map<Object, Object> clonedSystemProperties = (Map<Object, Object>) System.getProperties().clone();
+        Map<String, String> map = clonedSystemProperties.entrySet().stream()
+                .filter(entry -> String.valueOf(entry.getKey()).startsWith(INSTRUMENT_PREFIX))
```

Comment:
Should we filter out anything that isn't a string key or value instead, then cast?

Reply:
I'll add it, though since [`java.util.Properties` is a `Map<Object, Object>`](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html) that masquerades as a `Map<String, String>` and while there are ways to inject non-string keys/values, this is a relatively safe assumption here